Client/Application/AppContainer.swift

```diff
@@ -0,0 +1,75 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+/// TODO: Preparing services to be injected (https://mozilla-hub.atlassian.net/browse/FXIOS-4381) & actually
+/// injecting at those points (https://mozilla-hub.atlassian.net/browse/FXIOS-4152) will come in a later PR.
+
+import Foundation
+import os.log
+import Dip
+import Storage
+
+/// This is our concrete dependency container. It holds all dependencies / services the app would need through
+/// a session.
+class AppContainer: ServiceProvider {
+
+    static let shared: ServiceProvider = AppContainer()
+
+    /// The item holding registered services.
+    private var container: DependencyContainer?
+
+    init() {
+        container = bootstrapContainer()
+    }
+
+    /// Any services needed by the client can be resolved by calling this and passing the type.
+    func resolve<T>(type: T.Type) -> T {
+        do {
+            return try container?.resolve(T.self) as! T
+        } catch {
+            /// If a service we've thought was registered but can't be resolved, this is likely an issue within
+            /// bootstrapping. Double check your registrations and their types.
+            os_log(.error, "Could not resolve the requested type!")
```

Comment:
Instead of OS Log we have a logger in place you can use, look up `Logger.browserlogger` etc.

Reply:
I wanted to avoid coupling two frameworks like that, so I decided os_log would be good in this case. But if you think it's okay, I can definitely add that in in the upcoming PRs!